### ISSUE TITLE
 The code requires so much memory

### ISSUE THREAD

**Issue by shaolyy** [ChandlerBang/Pro-GNN#2]
Hi, 
When it runs  on google colab with  Tesla T4, I got  CUDA Out of Memory error message (Pubmed dataset). `Tried to allocate 1.45 GiB (GPU 0; 14.73 GiB total capacity; 13.08 GiB already allocated; 785.88 MiB free; 13.08 GiB reserved in total by PyTorch).`
I hope you kindly answer my following answers. 

Q1: Does the program need such memory? Is there any way to solve this problem?
Q2: On the pubmed dataset, I cannot reach the accuracy rate in the paper by GCN. Can you share the parameters or some suggestions?

Thanks.

**Comment by ChandlerBang** [comment]
Hi shaolyy,

Q1: I used GPU V100 (32GB) for Pubmed dataset since ProGNN can only deal with dense matrix. For other datasets, 12GB GPU Memory should be enough.

Q2: The parameter settings of GCN are `lr=0.01, weight_decay=5e-4, epoch=200, dropout=0.5` (same as that in the GCN paper). For the performance inconsistency,  I just checked my original implementation and found the difference is caused by the different data splits. In my experiments of Pubmed dataset, I mistakenly used `get_train_test_val(..., stratify=onehot_labels)` while it is `get_train_test_val(..., stratify=labels)`in other datasets. 

Now I've updated the code in `train.py`to generate the same data splits as in my experiments. Now you can update this repository and match the performance.
 
Thanks.

**Comment by shaoyf9** [comment]
Thank you very much for your reply, I will update the code.
